share/completions/git.fish

```diff
@@ -67,16 +67,24 @@ function __fish_git_remotes
 end
 
 function __fish_git_modified_files
+    set -l matching_files *(commandline -ct)*
+    if not set -q matching_files[1]
```

Comment:
I don't think this will ever be taken.

`commandline -ct` will print a single newline if the token is empty, resulting in one argument. That means this is effectively `set -l matching_files *""*` in that case, which will match any non-dot-file in the current directory (including directories).

If this, as it is now, is still fast enough for you, we can leave it. Otherwise we'd have to figure out something a bit smarter - e.g. ignoring directories (unless entered on the commandline).